packages/wdio-reporter/src/index.ts

```diff
@@ -207,19 +207,37 @@ export default class WDIOReporter extends EventEmitter {
             }
         })
 
+        /**
+         * deduplicate long script strings (> 1KB) to avoid memory leaks.
+         */
+        const scriptMap = new Map<string, string>()
+        const deduplicateScript =
+            (payload: { body: { script?: string } }) => {
+                const script = payload.body?.script
+                if (typeof script === 'string' && script.length > 1000) { // bytes
+                    const value = scriptMap.get(script)
```

Comment:
This means though that if the script is slightly different, it is still being accumulated in memory. I wonder if we just remove the script all together. If a reporter is interested in it, they can still capture it during the event.

Reply:
> If omitting them entirely would be too much, we could also do the same thing as what's done for the logs with `transformCommandLogResult`.

Let's do this.